packages/react-ai/src/components/AIConversation/context/AttachmentContext.tsx

```diff
@@ -1,16 +1,32 @@
 import * as React from 'react';
+import { AIConversationInput } from '../types';
 
-export const AttachmentContext = React.createContext(false);
+export interface AttachmentContextProps
+  extends Pick<
+    AIConversationInput,
+    'allowAttachments' | 'maxAttachments' | 'maxAttachmentSize'
+  > {}
+
+export const AttachmentContext = React.createContext<AttachmentContextProps>(
+  {}
+);
 
 export const AttachmentProvider = ({
   children,
-  allowAttachments,
-}: {
-  children?: React.ReactNode;
-  allowAttachments?: boolean;
-}): JSX.Element => {
+  allowAttachments = false,
+  // We save attachments as base64 strings into dynamodb for conversation history
+  // DynamoDB has a max size of 400kb for records
+  // This can be overridden so cutsomers could provide a lower number
+  // or a higher number if in the future we support larger sizes.
+  maxAttachmentSize = 400_000,
+  maxAttachments = 20,
```

Comment:
Are these safe to override today?

Reply:
They are safe to override to be lower, but if they are higher the customer will probably get an error further down in the stack (in lambda)

Reply:
Got it. Lmk if I missed it, but are we validating that the provided values are not larger than the known thresholds?

Reply:
We aren't validating if the provided values are larger than the known thresholds. I did this because the thresholds will go up at a later point, the 400kb size limit per file is a limitation of our backend implementation and not Bedrock or the AI model's, and we do plan to fix this long term. I can change it so we enforce the limits though if we feel strongly about it.